src/scripts/directives.js

```diff
@@ -350,24 +350,38 @@
 
                         var cast = (scope.useDuration) ? moment.duration : moment;
 
-                        scope.$watch('[hours, minutes]', function (newValues) {
-                            if (_.all(newValues, angular.isDefined)) {
-                                var values = _.object(_.map(group, function (key) {
-                                    return [key, scope[key] || 0];
-                                }));
+                        var setupModel = function () {
+                            if (angular.isDefined(scope.model)) {
+                                var value = cast(scope.model);
+                                var timeObject = {};
+                                var hours = group[0];
+                                var minutes = group[1];
+
+                                if (angular.isDefined(hours)) {
+                                    timeObject.hours = value.get(hours);
+                                }
+                                if (angular.isDefined(minutes)) {
+                                    timeObject.minutes = value.get(minutes);
+                                }
+                                angular.extend(scope, timeObject);
+                            }
+                        };
+
+                        setupModel();
```

Comment:
I'm just a bit wary, that if `scope.model` is resolved async, then it might not be available at the moment this directive is compiled. Maybe it would work using watch deregistrator? So keep the `$watch` on the model but assign it to a variable, and then inside `if (angular.isDefined(scope.model)) {}` call that variable - that will remove the watch. This way it will be resolved only when model is defined and we won't have a race condition - but still a loop will be avoided.